index.js

```diff
@@ -141,18 +141,24 @@ function makeCachedResolver (prefix, fieldName, cache, originalFieldResolver, po
     try {
       // dont use cache on mutation and subscriptions
       if (info.operation && (info.operation.operation === 'mutation' || info.operation.operation === 'subscription')) {
+        // TODO if originalFieldResolver throws, we should not go into the catch
         result = await originalFieldResolver(self, arg, ctx, info)
       } else if (
         (skip && (await skip(self, arg, ctx, info))) ||
         (policy && policy.skip && (await policy.skip(self, arg, ctx, info)))
       ) {
         // dont use cache on skip by policy or by general skip
         report[name].onSkip()
+        // TODO if originalFieldResolver throws, we should not go into the catch
         result = await originalFieldResolver(self, arg, ctx, info)
       } else {
         // use cache to get the result
         // Ignore execptions, 'onError' is already in place
-        result = await cache[name]({ self, arg, ctx, info }).catch(noop)
+        result = await cache[name]({ self, arg, ctx, info }).catch((err) => {
+          // TODO this should not be needed
+          err._onErrorCalled = true
+          throw err
```

Comment:
@sameer-coder can you check this out? Unfortunately I had to resort to this trick to make things work again. Can you  rework this code? I also added a lot of TODOs here that would need to be refactored.

@simone-sanfratello